[Azure AD] [ELM] created reprocess api (#13425)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accesspackageassignmentrequest-reprocess.md

---
title: "accessPackageAssignmentRequest: reprocess"
description: "Reprocess accessPackageAssignmentRequest objects."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackageAssignmentRequest: reprocess

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user’s request for access to an access package.

Reviewer: link to accessPackageAssignmentRequest resource type - https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentrequest?view=graph-rest-beta

> In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry an [accessPackageAssignment](../resources/accesspackageassignment.md) object of a user’s request for access to an access package.

Reviewer: let's make this "callers can automatically retry a user's request for access to an access package. It is performed on the [accessPackageAssignmentRequest](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentrequest?view=graph-rest-beta) object.

> In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user's request for access to an access package. It is performed on the [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md) object. 

Reviewer: - Users can't perform this action. Only admins with Access package assignment manager role (or higher) in ELM can perform this action.
- This action is most useful when there are failed and partially delivered requests within a 14-day time frame. It reduces friction since the user doesn't need to go to MyAccess and re-request the package. Instead, the admin can just reprocess the user's request in the Azure portal.

> In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user's request for access to an access package. It is performed on the [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md) object. 

Reviewer 2: ```suggestion
In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user's request for access to an access package. It is performed on an [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md) object whose **requestState** is in a `DeliveryFailed` or `PartiallyDelivered` state. 
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account) | EntitlementManagement.ReadWrite.All |
| Delegated (personal Microsoft account) | Not supported. |
| Application | EntitlementManagement.ReadWrite.All |
  
## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
```http
POST /identityGovernance/entitlementManagement/accessPackageAssignmentsRequests/{id}/reprocess  
```

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer \{token\}. Required. |

## Request body

Do not supply a request body for this method.

## Response

> If successful, this method returns a  `202 Accepted` response code. If the assignment doesn't exist, that is, the id isn't valid, this method will return a `400 bad request` or a `400 not found`response code.

Reviewer: this should say "If the assignment request doesn't exist..."

also, link to the request resource type again - https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentrequest?view=graph-rest-beta

> If successful, this method returns a  `202 Accepted` response code. If the assignment doesn't exist, that is, the id isn't valid, this method will return a `400 bad request` or a `400 not found`response code.

Reviewer: same as above - it will return 404 Not Found if the accessPackageAssignmentRequest object was not found and 400 Bad Request if the id isn't valid

> If successful, this method returns a  `202 Accepted` response code and retries the request of the [accessPackageAssignment](../resources/accesspackageassignment.md) object. If the assignment doesn't exist, this method will return `404 Not Found` or if the **id** isn't valid, this method returns a `400 Bad request` response code.

Reviewer: let's make this "...retries the request. If the [accessPackageAssignmentRequest](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageassignmentrequest?view=graph-rest-beta) object doesn't exist..."

> If successful, this method returns a  `202 Accepted` response code and retries the request of the [accessPackageAssignment](../resources/accesspackageassignment.md) object. If the assignment doesn't exist, this method will return `404 Not Found` or if the **id** isn't valid, this method returns a `400 Bad request` response code.

Reviewer: capitalize Request in Bad Request

> If successful, this method returns a  `202 Accepted` response code and retries the request. If the [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md) object doesn't exist, this method will return `404 Not Found` or if the **id** isn't valid, this method returns a `400 Bad request` response code.

Reviewer: capitalize Request in Bad Request and we're good to go!

---
title: "accessPackageAssignmentRequest: Reprocess"
description: "Reprocess accessPackageAssignmentRequest objects."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackageAssignmentRequest: Reprocess

Namespace: microsoft.graph

In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user’s request for access to an access package.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account) | EntitlementManagement.ReadWrite.All |
| Delegated (personal Microsoft account) | Not supported. |
| Application | EntitlementManagement.ReadWrite.All |
  
## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
```http
POST /identityGovernance/entitlementManagement/accessPackageAssignmentsRequests/{id}/reprocess  
```

> 
> ## Function parameters
> 
> None
> 
> ## Optional query parameters
> 
> None

Reviewer 2: Same comment
```suggestion
```

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer \{token\}. Required. |

## Request body

Do not supply a request body for this method.

## Response

> If successful, this method returns a `202 Accepted` response code. If the assignment doesn't exist, this method will return `404` or `400` if the ID is not valid.

Reviewer 2: If successful, this method returns a `202 Accepted` response code. If the assignment doesn't exist, that is, the **id** isn't valid, this method will return `400 Bad request` or a `400 Not found` response code.

## Examples

### Request

The following is an example of the request.

<!-- {
  "blockType": "ignored",
  "name": "reprocess_accesspackageassignmentsrequest"
}-->
```http
POST https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentRequests/d82eb508-acc4-43cc-bcf1-7c1c4a2c073b/reprocess
```

### Response

The following is an example of the response.

> > **Note:** The response object shown here might be shortened for readability.

Reviewer 2: Same comment
```suggestion
```

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account) | EntitlementManagement.ReadWrite.All |
| Application | EntitlementManagement.ReadWrite.All |
| Delegated (work or school account) | EntitlementManagement.ReadWrite.CurrentUser |
  
## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
```http
POST /identityGovernance/entitlementManagement/accessPackageAssignmentsRequests/{id}/reprocess  
```

## Function parameters

None

## Optional query parameters

None

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer \{token\}. Required. |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `202 Accepted` response code. If the assignment doesn't exist, this method will return `404` or `400` if the ID is not valid.

## Examples

### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "reprocess_accesspackageassignmentsrequest"
}-->
``` http

> POST https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentRequests/d82eb508-acc4-43cc-bcf1-7c1c4a2c073b/reprocess

Author: error wants GET instead of POST

```

### Response

The following is an example of the response.

> **Note:** The response object shown here might be shortened for readability.

<!-- {
  "blockType": "response",
  "truncated": true,
} -->

```http

> HTTP/1.1 202 Accepted 

Author: error is saying this format is wrong

---
title: "Reprocess on accessPackageAssignmentRequest"
description: "Reprocess accessPackageAssignmentRequest objects."
localization_priority: Normal
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# Reprocess on accessPackageAssignmentRequest

Namespace: microsoft.graph

In [Azure AD entitlement management](../resources/entitlementmanagement-root.md), callers can automatically retry a user’s request for access to an access package.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account) | EntitlementManagement.ReadWrite.All |
| Application | EntitlementManagement.ReadWrite.All |

> | Delegated (work or school account) | EntitlementManagement.ReadWrite.CurrentUser |

Author: are both work or school account?

> | Delegated (work or school account) | EntitlementManagement.ReadWrite.CurrentUser |

Reviewer 3: Remove .Current permission

  
## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
```http
POST /identityGovernance/entitlementManagement/accessPackageAssignmentsRequests/{id}/reprocess  
```

## Function parameters

None

## Optional query parameters

None

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer \{token\}. Required. |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `202 Accepted` response code. If the assignment doesn't exist, this method will return `404` or `400` if the ID is not valid.

## Examples

### Request

The following is an example of the request.

# [HTTP](#tab/http)
<!-- {
  "blockType": "request",
  "name": "reprocess_accesspackageassignmentsrequest"
}-->
``` http

> POST ttps://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignmentRequests/d82eb508-acc4-43cc-bcf1-7c1c4a2c073b/reprocess

Reviewer: We're missing the `h` in https here ^ @ajburnle
